### DISCUSSION TITLE
Android compilation error

### DISCUSSION THREAD

**Discussion by luojuntao** [rustdesk/rustdesk#5850] [Q&A]
https://rustdesk.com/docs/zh-cn/dev/build/android/

I referenced the content of the above link to compile, but the compilation failed and an error was reported. How to solve it?

luojt@54:~/rustdesk$ VCPKG_ROOT=$HOME/vcpkg ANDROID_NDK_HOME=$HOME/android-ndk-r23c flutter/ndk_arm64.sh
    Building arm64-v8a (aarch64-linux-android)
   Compiling magnum-opus v0.4.0 (https://github.com/rustdesk/magnum-opus#5cd2bf98)
   Compiling scrap v0.5.0 (/home/luojt/rustdesk/libs/scrap)
error: failed to run custom build command for `scrap v0.5.0 (/home/luojt/rustdesk/libs/scrap)`

Caused by:
  process didn't exit successfully: `/home/luojt/rustdesk/target/release/build/scrap-f01ab9d676072076/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-env-changed=NO_PKG_CONFIG_libyuv
  cargo:info=arm64-android
  cargo:rustc-link-lib=static=yuv
  cargo:rustc-link-search=/home/luojt/vcpkg/installed/arm64-android/lib
  cargo:include=/home/luojt/vcpkg/installed/arm64-android/include
  cargo:rerun-if-env-changed=NO_PKG_CONFIG_libvpx
  cargo:info=arm64-android
  cargo:rustc-link-lib=static=vpx
  cargo:rustc-link-search=/home/luojt/vcpkg/installed/arm64-android/lib
  cargo:include=/home/luojt/vcpkg/installed/arm64-android/include
  rerun-if-changed=/home/luojt/rustdesk/libs/scrap/src/bindings/vpx_ffi.h
  rerun-if-changed=/home/luojt/vcpkg/installed/arm64-android/include

  --- stderr
  thread 'main' panicked at '
  A `libclang` function was called that is not supported by the loaded `libclang` instance.

      called function = `clang_getTranslationUnitTargetInfo`
      loaded `libclang` instance = 3.8.x

  This crate only supports `libclang` 3.5 and later.
  The minimum `libclang` requirement for this particular function can be found here:
  https://docs.rs/clang-sys/latest/clang_sys/clang_getTranslationUnitTargetInfo/index.html

  Instructions for installing `libclang` can be found here:
  https://rust-lang.github.io/rust-bindgen/requirements.html
  ', /home/luojt/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clang-sys-1.6.1/src/lib.rs:1735:1
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to run custom build command for `magnum-opus v0.4.0 (https://github.com/rustdesk/magnum-opus#5cd2bf98)`

Caused by:
  process didn't exit successfully: `/home/luojt/rustdesk/target/release/build/magnum-opus-dc3e25aaaee13d45/build-script-build` (exit status: 101)
  --- stdout
  cargo:info=arm64-android
  cargo:rustc-link-lib=static=opus
  cargo:rustc-link-search=/home/luojt/vcpkg/installed/arm64-android/lib
  cargo:include=/home/luojt/vcpkg/installed/arm64-android/include
  rerun-if-changed=/home/luojt/.cargo/git/checkouts/magnum-opus-4bb999f3bcbf6ab0/5cd2bf9/opus_ffi.h
  rerun-if-changed=/home/luojt/vcpkg/installed/arm64-android/include

  --- stderr
  thread 'main' panicked at '
  A `libclang` function was called that is not supported by the loaded `libclang` instance.

      called function = `clang_Cursor_isMacroFunctionLike`
      loaded `libclang` instance = 3.8.x

  This crate only supports `libclang` 3.5 and later.
  The minimum `libclang` requirement for this particular function can be found here:
  https://docs.rs/clang-sys/latest/clang_sys/clang_Cursor_isMacroFunctionLike/index.html

  Instructions for installing `libclang` can be found here:
  https://rust-lang.github.io/rust-bindgen/requirements.html
  ', /home/luojt/.cargo/registry/src/index.crates.io-6f17d22bba15001f/clang-sys-1.6.1/src/lib.rs:1735:1
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
note: If the build failed due to a missing target, you can run this command:
note: 
note:     rustup target install aarch64-linux-android

Can anyone help me solve it?

**Comment by rustdesk** [comment]
#763
